src/manifest.c

```diff
@@ -518,9 +518,11 @@ static int retrieve_manifests(int current, int version, char *component, struct
 		goto out;
 	}
 
-	if (!signature_download_and_verify(url, filename)) {
-		unlink(filename);
-		goto out;
+	if (signing_enabled) {
+		if (!signature_download_and_verify(url, filename)) {
+			unlink(filename);
+			goto out;
+		}
```

Comment:
Though the signature-related function calls sites have not been updated yet, the only Manifest.MoM.tar will be signed (out of the manifest tar currently published). Can you add an extra check to ensure is the MoM manifest?